### PR TITLE
feat(api): add model for club registrations

### PR DESCRIPTION
**Pull request by PranshuNayak**

### CODE REVIEW

**Review by Samy-33 — APPROVED** [review]
LGTM

File: api/api/clubs/models.py [hunk 1]
```diff
@@ -17,3 +20,26 @@ class Club (models.Model):
...
+class ClubRegistrationRequests(models.Model):
+    STATUS_PENDING = 'Pending'
+    STATUS_APPROVED = 'Approved'
+    STATUS_REJECTED = 'Rejected'
+
+    STATUS_CHOICES = [
+        ('Pending', STATUS_PENDING),
+        ('Approved', STATUS_APPROVED),
+        ('Rejected', STATUS_REJECTED)
+    ]
```
[the remark this answers is not in the document]
Comment (PranshuNayak) [reply api/api/clubs/models.py:34]: This looks better [https://docs.djangoproject.com/en/4.0/ref/models/fields/#enumeration-types](url)

[the remark this answers is not in the document]
Comment (Samy-33) [reply api/api/clubs/models.py:34]: Sure, looks good. Your link is broken tho.

[the remark this answers is not in the document]
Comment (PranshuNayak) [reply api/api/clubs/models.py]: made the changes

File: api/api/clubs/models.py [hunk 2]
```diff
@@ -17,3 +21,22 @@ class Club (models.Model):
...
     logo = models.URLField(max_length=500)
+
+
+class ClubRegistrationRequests(models.Model):
+    class RegistrationStatus(models.TextChoices):
+        PENDING = 'Pending', _('Pending')
+        APPROVED = 'Approved', _('Approved')
+        REJECTED = 'Rejected', _('Rejected')
+
+    user = models.ForeignKey(User, related_name='registered_clubs', on_delete=models.CASCADE)
```
Comment (Samy-33) [inline api/api/clubs/models.py:32]: call the related name as `registration_requests`

File: api/api/clubs/models.py [hunk 3]
```diff
@@ -17,3 +21,22 @@ class Club (models.Model):
...
+    club = models.ForeignKey(Club, related_name='registration_requests', on_delete=models.CASCADE)
+    fee_submitted = models.BooleanField(default=False)
+    status = models.CharField(max_length=10, choices=RegistrationStatus.choices,
+                              default=RegistrationStatus.PENDING)
+    remark = models.TextField(blank=True)
+    updated_by = models.ForeignKey(
+        User, related_name='updated_club_registration_requests',
```
Comment (Samy-33) [inline api/api/clubs/models.py:39]: too big related name?

**Review by Samy-33 — CHANGES_REQUESTED** [review]
looks good overall

File: api/api/clubs/models.py (hunk 3 above)
Comment (PranshuNayak) [reply api/api/clubs/models.py:39]: Tried to give a meaningful name . May be changed to `updated_registration_requests` or `updated_requests` ? But they lack context I think . Any suggestion ?

Comment (Samy-33) [reply api/api/clubs/models.py:39]: `last_updated_requests`

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**